data_prep/tests/test_create_climo_files.py

```diff
@@ -0,0 +1,205 @@
+
+
+@mark.parametrize('input_and_climo_files, t_start, t_end', [
+    (('gcm', False, t_start(1965), t_end(1970)), t_start(1965), t_end(1970)),
+    (('downscaled_tasmax', False, t_start(1961), t_end(1990)), t_start(1961), t_end(1990)),
+    (('downscaled_pr', False, t_start(1961), t_end(1990)), t_start(1961), t_end(1990)),
+    (('hydromodel_gcm', True, t_start(1984), t_end(1995)), t_start(1984), t_end(1995)),
+], indirect=['input_and_climo_files'])
+def test_climo_metadata(input_and_climo_files, t_start, t_end):
+    """Test that the correct climo-specific metadata has been added/updated."""
+    input_file, climo_files = input_and_climo_files
+    for fp in climo_files:
+        with CFDataset(fp) as cf:
+            assert cf.is_multi_year_mean
+            assert cf.frequency == {
+                'daily': 'msaClim',
+                'monthly': 'saClim',
+                'yearly': 'aClim'
+            }[input_file.time_resolution]
+            assert cf.climo_start_time == t_start.isoformat()[:19] + 'Z'
+            assert cf.climo_end_time == t_end.isoformat()[:19] + 'Z'
+            assert getattr(cf, 'climo_tracking_id', None) == getattr(input_file, 'tracking_id', None)
+            if 'pr' in cf.dependent_varnames:
+                assert cf.variables['pr'].units.endswith('d-1')
+
+
+@mark.parametrize('input_and_climo_files', [
+    ('gcm', False, t_start(1965), t_end(1970)),
+    ('downscaled_tasmax', False, t_start(1961), t_end(1990)),
+    ('downscaled_pr', False, t_start(1961), t_end(1990)),
+    ('hydromodel_gcm', False, t_start(1984), t_end(1995)),
+    ('hydromodel_gcm', True, t_start(1984), t_end(1995)),
+], indirect=['input_and_climo_files'])
+def test_dependent_variables(input_and_climo_files):
+    """Test that the output files contain the expected dependent variables"""
+    input_file, climo_files = input_and_climo_files
+    dependent_varnames_in_cfs = set()
+    for fp in climo_files:
+        with CFDataset(fp) as cf:
+            dependent_varnames_in_cfs.update(cf.dependent_varnames)
+            if len(climo_files) > 1:
+                # There should be one dependent variable from the input file
+                assert len(cf.dependent_varnames) == 1
+    # All the input dependent variables should be covered by all the output files
+    assert dependent_varnames_in_cfs == set(input_file.dependent_varnames)
+
+
+@mark.parametrize('input_and_climo_files, t_start, t_end', [
+    (('gcm', False, t_start(1965), t_end(1970)), t_start(1965), t_end(1970)),
+    (('downscaled_tasmax', False, t_start(1961), t_end(1990)), t_start(1961), t_end(1990)),
+    # No need to repleat with downscaled_pr
+    (('hydromodel_gcm', False, t_start(1984), t_end(1995)), t_start(1984), t_end(1995)),
+    (('hydromodel_gcm', True, t_start(1984), t_end(1995)), t_start(1984), t_end(1995)),
+], indirect=['input_and_climo_files'])
+def test_time_and_climo_bounds_vars(input_and_climo_files, t_start, t_end):
```

Comment:
I'd really like to use a simpler computation (and tests) if there is no reason to be so complicated.